packages/embark/src/cmd/cmd_controller.js

```diff
@@ -504,58 +504,73 @@ class EmbarkController {
+        engine.registerModuleGroup("filewatcher");
+        engine.registerModuleGroup("storage");
+        engine.registerModuleGroup("communication");
+        engine.registerModulePackage('embark-deploy-tracker', {plugins: engine.plugins});
+        engine.registerModulePackage('embark-plugin-cmd', {embarkConfigFile: engine.embarkConfig, embarkConfig: engine.config.embarkConfig, packageFile: 'package.json'});
+        callback();
+      },
+      callback => {
         if (isSecondaryProcess(engine)) {
           return callback();
         }
-        engine.startService("embarkListener");
-        engine.startService("processManager");
-        engine.startService("coreProcess");
-        engine.startService("serviceMonitor");
-        engine.startService("libraryManager");
-        engine.startService("pipeline");
-        engine.startService("storage");
-        engine.startService("cockpit");
-        engine.startService("pluginCommand");
 
-        engine.events.request('blockchain:ready', callback);
-      },
-      function ipcConnect(callback) {
-        return callback();
-      },
-      function deploy(callback) {
-        // Skip if we are connected to a websocket, the server will do it
-        if(isSecondaryProcess(engine)) {
-          return callback();
-        }
-        engine.config.reloadConfig();
-        engine.events.request('deploy:contracts', function (err) {
-          callback(err);
+        let plugin = engine.plugins.createPlugin('cmdcontrollerplugin', {});
+        plugin.registerActionForEvent("embark:engine:started", async (_params, cb) => {
+          await engine.events.request2("blockchain:node:start", engine.config.blockchainConfig, cb);
         });
-      },
-      function waitForWriteFinish(callback) {
-        // Skip if we are connected to a websocket, the server will do it
-        if(isSecondaryProcess(engine)) {
-          return callback();
-        }
-        engine.logger.info("Finished deploying".underline);
-        engine.events.once('outputDone', (err) => {
-          engine.logger.info(__("finished building").underline);
-          callback(err);
+        plugin.registerActionForEvent("embark:engine:started", async (_params, cb) => {
+          await engine.events.request2("storage:node:start", engine.config.storageConfig, cb);
+        });
+        plugin.registerActionForEvent("embark:engine:started", async (_params, cb) => {
+          await engine.events.request2("communication:node:start", engine.config.communicationConfig, cb);
+        });
+
+        engine.events.on('outputDone', function () {
+          engine.logger.info((__("Looking for documentation? You can find it at") + " ").cyan + "http://embark.status.im/docs/".green.underline + ".".cyan);
+          engine.logger.info(__("Ready").underline);
+          engine.events.emit("status", __("Ready").green);
+        });
+
+        engine.events.on('file-event', async ({ fileType, path }) => {
```

Comment:
I think we only need the file-events when the console is started stand-alone (not next to another embark run instance.
This might already be handled by the engine or the modules, just asking in case we need to consider it here.

Reply:
Check the old line 552 to see what I mean

Reply:
It's a little bit hard to see from here, but further up in the function we return early if `isSecondaryProcess()` which achieves exactly that.